### ISSUE TITLE
best model

### ISSUE THREAD

**Issue by ai8049520** [roboflow/rf-detr#16]
Hello,  I try train rfdet 100 epochs on coco dataset with 1 class, and
get good metrics ({"class_map": {"valid": [{"map95": 0.7727289854838079, "map50": 0.9457314068264822, "class": "all"}]}}) and have basic questions - 
I have 1 class and it indexed with 0 (not 1 as coco) is this correct?:
    "info": {},
    "licenses": [],
    "categories": [
        {
            "id": 0,
            "name": "Mhp"
        }
    ],

I get best model on path "/content/output/checkpoint_best_total.pth" but there is also other candidate for "best" -  checkpoint_best_regular.pth and checkpoint_best_ema.pth and checkpoint.pth. Where is best model for best metric value? I read in blog that 50 epochs is good start for model training ( I choose 100). My question is there any documentation (or plans for it)  for hyper params settings, augmentation and optimization? How resume from checkpoint, and other basic staff? Thank you!

**Comment by probicheaux** [comment]
These are great questions and we definitely have more work to do documenting the things you've mentioned.

> I get best model on path "/content/output/checkpoint_best_total.pth" but there is also other candidate for "best" - checkpoint_best_regular.pth and checkpoint_best_ema.pth and checkpoint.pth. Where is best model for best metric value?

During training, we essentially create 2 models -- the regular model that you're training, and another version called the EMA model, which is an average of all model checkpoints. The best regular is saved to `checkpoint_best_regular.pth` and the best EMA is saved to `checkpoint_best_ema.pth`. At the end of training we see who got the all time highest performance on the val set and copy over to `checkpoint_best_total.pth`, so that one should be used for downstream inference.

> How resume from checkpoint
```python
from rfdetr import RFDETRBase
model = RFDETRBase(pretrain_weights="your/pretrain/weights.pth", resume="your/pretrain/weights.pth")
model.train(...)
```

> augmentation

this isn't currently configurable


> optimization

we haven't measured anything except for the defaults

**Comment by ai8049520** [comment]
Thank you!
